ftdetect/neuron.vim

```diff
@@ -1,20 +1,15 @@
 let g:neuron_extension = get(g:, 'neuron_extension', '.md')
-let g:neuron_dir = get(g:, 'neuron_dir', fnamemodify(expand("%:p"), ":h")."/")
-
-" fallback to using getcwd if the above gives us a relative path
-if g:neuron_dir == './'
-	let g:neuron_dir = getcwd() . "/"
-endif
+let b:neuron_dir = get(g:, 'neuron_dir')
 
 " if there is no neuron.dhall file in current dir then it is not a zettelkasten
-if !filereadable(g:neuron_dir."neuron.dhall")
+if !filereadable(b:neuron_dir."neuron.dhall")
 	finish
 endif
 
 if exists('b:did_ftdetect') | finish | endif
 aug neuron
-	exec ':au! BufRead '.fnameescape(g:neuron_dir).'*'.fnameescape(g:neuron_extension).' call neuron#add_virtual_titles()'
-	exec ':au! BufEnter '.fnameescape(g:neuron_dir).'*'.fnameescape(g:neuron_extension).' call neuron#on_enter()'
-	exec ':au! BufWrite '.fnameescape(g:neuron_dir).'*'.fnameescape(g:neuron_extension).' call neuron#on_write()'
+	exec ':au! BufRead '.fnameescape(b:neuron_dir).'*'.fnameescape(b:neuron_extension).' call neuron#add_virtual_titles()'
+	exec ':au! BufEnter '.fnameescape(b:neuron_dir).'*'.fnameescape(b:neuron_extension).' call neuron#on_enter()'
+	exec ':au! BufWrite '.fnameescape(b:neuron_dir).'*'.fnameescape(b:neuron_extension).' call neuron#on_write()'
```

Comment:
Should b:neuron_extension have remained g:neuron_extension? Since this change I'm getting the following error:
Error detected while processing /home/daan/.local/share/nvim/plugged/neuron.vim/ftdetect/neuron.vim:
line   11:
E121: Undefined variable: b:neuron_extension
E116: Invalid arguments for function fnameescape
line   12:
E121: Undefined variable: b:neuron_extension
E116: Invalid arguments for function fnameescape
line   13:
E121: Undefined variable: b:neuron_extension
E116: Invalid arguments for function fnameescape

Reply:
@dspijkers could you please give me a list of steps that you followed to see that error?

Reply:
Ah yeah of course sorry.  I updated to the latest version using vim-plug, and now if I define neuron_dir inside my neovimrc then I get this error every time I start up neovim.

Reply:
interesting - if I set it in my rc I don't get the error. Could you please tell me:
- which directory are you starting from?(zettelkasten directory or anotther?
- what exactly do you `neuron_dir` set to and how are you setting it? after or before `Plug `fiatjaf/neuron.vim`?
- are you starting neovim with no file specified or with a file specified?

Reply:
Sure!

- The directory doesn't matter, it causes an error both inside zettelkasten and outside
- The exact setting I have is: let g:neuron_dir = "home/daan/notes/"
  If I have it before loading the plugin then it causes an error, but if I place if after then everything works fine.
- both with and without a file specified it gives an error.

So I only get it if I set it before loading the plugin.

Reply:
right, yeah, putting it before the plugin load, that would be it. I'll open up a separate ticket around it.

Reply:
having said that, I still can't reproduce the problem. If I set it before, I don't get an error. I guess for now just put it after and we'll see if it affects anything else.

Reply:
For what it's worth, I have the same issue too after updating. And isn't `neuron_dir` unrelated to this particular problem? The error seems to result from `b:neuron_extension` being referenced in the lines above while the assignment on line 1 is to `g:neuron_extension`:

https://github.com/fiatjaf/neuron.vim/blob/853fbe273f9f40defe69a9d50d267fee1bbc6b5a/ftdetect/neuron.vim#L1

So it doesn't seem that `b:neuron_extension` isn't actually defined anywhere, hence the error? Let me know if I'm missing something though :slightly_smiling_face:

Reply:
....right. Totally missed that it was `extension` and not `dir`. Will sort that out. Thanks @pauljxtan

Reply:
Pushed a fix. Thanks @pauljxtan.